bach/bach/dataframe.py

```diff
@@ -908,6 +882,61 @@ def __setitem__(self,
         else:
             raise ValueError(f'Key should be either a string or a list of strings, value: {key}')
 
+    def _index_merge(self,
+                     key: str,
+                     value: 'Series',
+                     how: str = 'left'):
+        """"
+        Internal method used by __setitem__ to set a series using a merge on index. Modifies the DataFrame
+        with the added column. The DataFrames index name is the same as the original DataFrame's.
+
+        :param key: name of the column to set.
+        :param value: Series that is set.
+        :param how: 'left' or 'outer'.
+        """
+
+        # todo This method's functionality will be implementd in merge()
+        if not (len(value.index) == 1 and len(self.index) == 1):
+
+            raise ValueError('setting with different base nodes only supported for one level'
+                             ' index')
+        index_name = self.index_columns[0]
+        value_index_name = list(value.index.keys())[0]
+        if self.index[index_name].dtype != value.index[value_index_name].dtype:
+            raise ValueError('dtypes of indexes should be the same')
+        renamed_value = value.copy_override(name=key)
+
+        if how == 'outer':
+            renamed_value = renamed_value.to_frame().materialize()
+            renamed_value[value_index_name + '__index'] = renamed_value.index[value_index_name]
+
+        df = self.merge(renamed_value,
+                        left_index=True,
+                        right_index=True,
+                        how=how,
+                        suffixes=('', '__remove'))
+
+        if how == 'outer':
+            new_index = df.index[index_name].fillna(df[value_index_name + '__index'])
+        elif how == 'left':
+            new_index = df.index[index_name]
+        else:
+            raise NotImplementedError(f'how "{how}" not supported')
```

Comment:
I think it's not necessary to raise `NotImplementedError`. This should be handled by merge operation.
```suggestion
         new_index = df.index[index_name]
        if how == 'outer':
            new_index = new_index.fillna(df[value_index_name + '__index'])
```

Reply:
this function is specifically only to be used with outer and left, and only for `__setitem__` and `_get_supported` from series.py. I didn't test other situations. Now if you would do 'right' for example with you suggested change, it would give only one index from the 'left' df (and the data from the index from the right would be missing). 
Therefore suggest to leave it as it is, but also move the functionality to merge asap, so this function can be removed.